async_pi_estimate/node-addon-api/async.cc

```diff
@@ -0,0 +1,39 @@
+#include <napi.h>
+#include "pi_est.h"  // NOLINT(build/include)
+#include "async.h"  // NOLINT(build/include)
+
+class PiWorker : public Napi::AsyncWorker {
+ public:
+  PiWorker(Napi::Function& callback, int points)
+    : Napi::AsyncWorker(callback), points(points), estimate(0) {}
+  ~PiWorker() {}
+
+  // Executed inside the worker-thread.
+  // It is not safe to access V8, or V8 data structures
+  // here, so everything we need for input and output
+  // should go on `this`.
+  void Execute () {
+    estimate = Estimate(points);
+  }
+
+  // Executed when the async work is complete
+  // this function will be run inside the main event loop
+  // so it is safe to use V8 again
```

Comment:
probably don't want the reference to V8 here.

Reply:
Replaced V8 with JS engine